### DISCUSSION TITLE
Can I break the layer lock when converting to psd?

### DISCUSSION THREAD

**Discussion by la-darknesss** [ImageMagick/ImageMagick#6228] [Help]
### ImageMagick version

7.1.0-Q16-HDRI

### Operating system, version and so on

windows10

### Description

You can convert a jpg file to a psd file with the following command
`magick.exe png_01.png output.psd`

By default, there is a layer lock.
Is there an option to remove the layer lock?

![ss](https://user-images.githubusercontent.com/129155371/230020908-978c6755-6fe6-472b-84c4-f21b8352f0c3.png)

**Comment by fmw42** [comment]
As far as I know, there is no current option to do that in Imagemagick.  Why is that important?

**Reply by la-darknesss** [reply]
Thank you for every reply.
Not for systemic reasons, but when editing a psd file generated by ImageMagick, I always release all layer locks.
I asked this question because I thought it would be convenient to eliminate the need to unlock them.

**Reply by fmw42** [reply]
I will leave it to one of the IM developers to comment further.

**Reply by la-darknesss** [reply]
Yes, thank you.

**Reply by dlemstra** [reply]
We are using the following information: https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/. I am not sure what enables this but I suspect its the `transparency protected` flag. Is the layer locked when you open this file: [output.zip](https://github.com/ImageMagick/ImageMagick/files/11180049/output.zip)?

**Reply by fmw42** [reply]
Looks to me that the layer is locked in that file when I open it in Photoshop.

**Reply by dlemstra** [reply]
Can you toggle the flag and save the file? Maybe I can check the difference with a hex editor?

**Comment by fmw42** [comment]
Both files:

[Archive.zip](https://github.com/ImageMagick/ImageMagick/files/11180309/Archive.zip)

**Comment by dlemstra** [comment]
It looks like they set the layer flags to `8`. According to the "documentation" this means "pixel data irrelevant to appearance of document" but I have I no clue what that means. What happens when you open this file: [output.zip](https://github.com/ImageMagick/ImageMagick/files/11182779/output.zip)

**Reply by fmw42** [reply]
Still locked in Photoshop

**Reply by la-darknesss** [reply]
＠dlemstra, fmw42
Thank you for your assistance.
As confirmed by fmw42.
I checked the attached file in PhotoShop and it is layer locked.

**Reply by dlemstra** [reply]
Well I have no clue where this information is stored so I don't think we can add something that disables locking the layer.

**Reply by la-darknesss** [reply]
Thank you for your research.
I also checked the parameters on my end, but could not find any...
If you notice anything else, please leave a comment.
